doc/development/adding_operators.rst

```diff
@@ -0,0 +1,357 @@
+            # trainable inputs could be tensors of different types
+            shape = qml.math.shape(angle)
+            if len(shape) > 1:
+                raise ValueError(f"Expected a scalar angle; got angle of shape {shape}.")
+
+            #------------------------------------
+
+            # do_flip is not trainable but influences the action of the operator,
+            # which is why we define it to be a hyperparameter
+            self._hyperparameters = {
+                "do_flip": do_flip
+            }
+
+            # we extract all wires that the operator acts on,
+            # relying on the Wire class arithmetic
+            all_wires = qml.wires.Wires(wire_rot) + qml.wires.Wires(wire_flip)
+
+            # The parent class expects all trainable parameters to be fed as positional
+            # arguments, and all wires acted on fed as a keyword argument.
+            # The id keyword argument allows users to give their instance a custom name.
+            # The do_queue keyword argument specifies whether or not
+            # the operator is queued when created in a tape context.
+            super().__init__(angle, wires=all_wires, do_queue=do_queue, id=id)
+
+        @property
+        def num_params(self):
+            # if it is known before creation, define the number of parameters to expect here,
+            # which makes sure an error is raised if the wrong number was passed
+            return 1
+
+        @staticmethod
+        def compute_decomposition(angle, wires, do_flip):  # pylint: disable=arguments-differ
+            # Overwriting this method defines the decomposition of the new gate, as it is
+            # called by Operator.decomposition().
+            # The general signature of this function is (*parameters, wires, **hyperparameters).
+            op_list = []
+            if do_flip:
+                op_list.append(qml.PauliX(wires=wires[1]))
+            op_list.append(qml.RX(angle, wires=wires[0]))
+            return op_list
+
+        def adjoint(self):
+            # the adjoint operator of this gate simply negates the angle
+            return FlipAndRotate(-self.parameters[0], self.wires[0], self.wires[1], do_flip=self.hyperparameters["do_flip"])
+
+The new gate can now be created as follows:
+
+>>> op = FlipAndRotate(0.1, wire_rot="q3", wire_flip="q1", do_flip=True)
+>>> op
+FlipAndRotate(0.1, wires=['q3', 'q1'])
+>>> op.decomposition()
+[PauliX(wires=['q1']), RX(0.1, wires=['q3'])]
+>>> op.adjoint()
+FlipAndRotate(-0.1, wires=['q3', 'q1'])
+
+The new gate can be used with PennyLane devices. PennyLane checks with the device
+whether it supports operations using the operation name.
+
+- If the device registers support for an operation with the same name,
+  PennyLane leaves the gate implementation up to the device. The device
+  might have a hardcoded implementation, _or_ it may refer to one of the
+  numerical representations of the operator (such as `operator.matrix`).
```

Comment:
```suggestion
- If the device registers support for an operation with the same name,
  PennyLane leaves the gate implementation up to the device. The device
  might have a hardcoded implementation, *or* it may refer to one of the
  numerical representations of the operator (such as :meth:`.Operator.matrix`).
```